test/tests/functional/pbs_ralter.py

```diff
@@ -0,0 +1,1007 @@
+                 'reserve_duration': duration}
+        self.server.expect(RESV, attrs, id=rid)
+
+    def submit_job_to_resv(self, rid, sleep=10):
+        """
+        Helper method for submitting a sleep job to the reservation.
+
+        param rid: Reservation id.
+        type  rid: string.
+
+        param sleep: Sleep time in seconds for the job.
+        type  sleep: int.
+        """
+        r_queue = rid.split('.')[0]
+        a = {'queue': r_queue}
+        j = Job(TEST_USER, a)
+        j.set_sleep_time(sleep)
+        jid = self.server.submit(j)
+        return jid
+
+    def alter_a_reservation(self, rid, start, end, shift,
+                            alter_s=False, alter_e=False,
+                            ExpectSuccess=1, confirm=True,
+                            interactive=0, sequence=1):
+        """
+        Helper method for altering a reservation.
+        This method also checks for the server and accounting logs.
+
+        param rid: Reservation id.
+        type  rid: string.
+
+        param start: Start time of the reservation.
+        type  start: int.
+
+        param end: End time of the reservation.
+        type  end: int.
+
+        param shift: Time in seconds the reservation times will be moved.
+        type  shift: int.
+
+        param alter_s: Whether the caller intents to change the start time.
+                       Default - No.
+        type  alter_s: bool.
+
+        param alter_e: Whether the caller intents to change the end time.
+                       Default - No.
+        type  alter_e: bool.
+
+        param ExpectSuccess: Whether the caller expects the reservation
+                             to be successful or not.
+                             Default: Yes.
+        type  ExpectSuccess: int.
+
+        param confirm: The expected state of the reservation after it is
+                       altered. It can be either Confirmed or Running.
+                       Default - Confirmed State.
+        type  confirm: bool.
+
+        param interactive: Time in seconds the CLI waits for a reply.
+                           Default - 0 seconds.
+        type  interactive: int.
+
+        param sequence: To check the number of logs corresponding to alter.
+                        Default: 1
+        type  sequence: int.
+        """
+        duration = end - start
+        new_start = start
+        new_end = end
+        flag = 0
+        attrs = {}
+        bu = BatchUtils()
+        standing = 0
+
+        if (alter_s):
+            new_start = start + shift
+            new_start_conv = bu.convert_seconds_to_resvtime(
+                new_start)
+            attrs.update({'reserve_start': new_start_conv})
+
+        if (alter_e):
+            new_end = end + shift
+            new_end_conv = bu.convert_seconds_to_resvtime(new_end)
+            attrs.update({'reserve_end': new_end_conv})
+
+        if (interactive > 0):
+            attrs.update({'interactive': interactive})
```

Comment:
depending on which boolean is set, the values get added to the dictionary. I find this intuitive.